packages/client/tests/functional/dataproxy-engine/version/tests.ts

```diff
@@ -0,0 +1,21 @@
+import testMatrix from './_matrix'
+// @ts-ignore
+import type { PrismaClient } from './node_modules/@prisma/client'
+
+declare let prisma: PrismaClient
+
+testMatrix.setupTestSuite(() => {
```

Comment:
I am not 100% sure we should assert on private properties here. Test below (the one that uses the extension) already verifies that correct header is sent through more public means. Do you think we still need this one?

Reply:
It's also testing `version()` which is not covered by the other test. though `version()` is actually only used here.

So I think it's somewhat interesting to have this, though I agree testing private properties is not great it's maybe better than nothing.
Here it also shows how `remoteClientVersion` is undefined before calling `$connect()`.